### PR TITLE
net-http spy: use Module.prepend instead of alias

### LINKED ISSUE #379: spies should use Module.prepend vs alias
Using `Module.prepend` is the default to override a method in Rails. Prior to that `alias_method_chain`.

We've been trying to use the APM with [Foreman](https://github.com/theforeman/foreman), which [overrides](https://github.com/theforeman/foreman/blob/b7d88a6c5c307312fae0404ceb1d5717e1cf313a/lib/foreman/http_proxy/net_http_extension.rb#L22) `net-http`'s `request` method as well as apm-agent-ruby. Foreman uses `Module.prepend`, [this gem](https://github.com/elastic/apm-agent-ruby/blob/87f9d2eaa3f9001fb120478ae4e7035b22be0527/lib/elastic_apm/spies/net_http.rb#L33) uses `alias`. And this leads to a `SystemStackError (stack level too deep)` exception.

As a workaround we currently disable the net-http spy in the config:
```yaml
disabled_spies:
  - json
  - net_http
```

Can we switch to `Module.prepend` as a default or make this configurable?

### PR DESCRIPTION
**Pull request by timogoebel**
This partially fixes #379.

### CODE REVIEW

File: lib/elastic_apm/spies/net_http.rb [hunk 1]
```diff
@@ -28,34 +28,34 @@ def disable_in
         end
       end
 
-      def install
-        Net::HTTP.class_eval do
-          alias request_without_apm request
+      module Overrides
```
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/spies/net_http.rb]: Just put a `# @api private` above this to ignore it

**Comment by mikker** [comment]
_Partially_ because of the Faraday follow-up?

Thanks a bunch for this!

**Comment by mikker** [comment]
jenkins, retest this please

**Comment by timogoebel** [comment]
> _Partially_ because of the Faraday follow-up?

yep, ideally, we'd use `Module.prepend` for every spy.

**Comment by mikker** [comment]
Yes, that would definitely be ideal. I'll add a separate issue to convert all of the remaining spies.

**Comment by mikker** [comment]
Scratch that – that's already what your issue is about 😆

**Comment by mikker** [comment]
💙💚💛💜❤️

**Pull request merged**